test/test-async.js

```diff
@@ -1889,8 +1890,9 @@ exports['map'] = {
     var r = [];
     async.map(a, function(x, callback){
         r.push(x);
+        var done = r.length == a.length;
         callback(null);
-        if (r.length >= a.length) {
+        if (done) {
             test.same(r, a);
```

Comment:
I've created a 2.x branch we could merge this PR into.  Although, I'm a bit hesitant because it could set us up for merge hell  if we end up making lots of changes on master.